### PR TITLE
[BEAM-2699] Update AppliedPTransform Equality

### PR DESCRIPTION
**Pull request by tgroh**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/projects/BEAM/issues/) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue.
 - [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---
Use only node name and the pipeline the two transforms are in. Do not
consider the reported equality of inputs, outputs, or transforms.

### CODE REVIEW

**Comment by tgroh** [comment]
R: @jkff @kennknowles (Would like both here)

File: sdks/java/core/src/main/java/org/apache/beam/sdk/runners/AppliedPTransform.java [hunk 1]
```diff
@@ -70,4 +70,19 @@
   public abstract TransformT getTransform();
 
   public abstract Pipeline getPipeline();
+
+  @Override
+  public boolean equals(Object other) {
```
Comment (kennknowles) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/runners/AppliedPTransform.java]: I'm going to say the same thing you said to me - probably the map should be keyed by the stable name or an equivalence wrapper, so the object itself can have context-unaware equality.

**Comment by tgroh** [comment]
I am for doing it, but have not found the time. I'm happy to close this in the intermediate time

**Comment by kennknowles** [comment]
OK, going to close to increase signal for more active PRs. Feel free to reopen whenever you are ready.

**Pull request closed without merging**